### DISCUSSION TITLE
Help me understand HW and models for Frigate 16

### DISCUSSION THREAD

**Discussion by szaske** [blakeblackshear/frigate#19899] [Ask A Question]
I've been running and loving Frigate for a while now, and just updated my server hardware.  I've been using a USB coral this whole time, but would prefer something internal instead now that I'm upgrading my server. 

I'm confused about the status of Coral. With all the excitement about AI I would think Coral...which if I remember correctly was a unit of Google would be a hot company.  But for all appearances Coral is dead. The website hasn't been updated in years.  Can someone explain why we haven't really seen ANY good replacements for the Coral HW? My trusty old USB Coral inference times are 8-9ms (with 4 TOPs) . The latest Frigate documentation shows inference times and it appear the only competitor close to that is the Hailo device. The 8 seems impossible to find for sale and the 8L sells for around $70. I can buy a 2x Coral M2 for $55.  It seems weird to be buying such an old device in 2025 from a company that seems to be out of business.  Am I missing something?  

I haven't switched to Frigate 16 yet and perhaps someone here can explain to me what the difference is between all these different "models", or why I'd want one over the other. My setup is Coral USB for detection and old K620 video card to help with video decoding. I have 4 3840x2160 cameras.  It sounds like I'm using the TensorFlow Lite model for detection.  When I switch to Frigate 16 would the Coral be used for face detection as well or do I need to switch to another model for face detection?

Is there any benefit to switching to a Hailo 8L or potentially an Intel Arc A380 instead of just buying a Coral 2xM2?

**Comment by inkeep-ai-assistant** [comment]
For Frigate v0.16, several hardware options are supported for object detection:

**Coral (USB, M.2, PCIe):**
- Coral EdgeTPU is still fully supported and compatible with Frigate v0.16, both in USB and M.2/PCIe formats.
- Inference times for Coral are typically in the 5–10 ms range per frame[(1)](https://docs.frigate.video/frigate/hardware).
- To use a Coral device internally, you can use the M.2 or PCIe version. Both are supported with the following configuration:
    ```yaml
    detectors:
      coral:
        type: edgetpu
        device: pci
    ```
    Or for multiple M.2/PCIe Corals:
    ```yaml
    detectors:
      coral1:
        type: edgetpu
        device: pci:0
      coral2:
        type: edgetpu
        device: pci:1
    ```
[(2)](https://docs.frigate.video/configuration/object_detectors)

**Hailo 8/8L:**
- Hailo-8 and Hailo-8L are supported as internal accelerators (M.2, PCIe). The Hailo-8L is more widely available and affordable.
- Inference times for Hailo-8L are around 11 ms per frame, and for Hailo-8 about 7 ms per frame on the default YOLOv6n model[(1)](https://docs.frigate.video/frigate/hardware).
- Configuration for Hailo looks like:
    ```yaml
    detectors:
      hailo:
        type: hailo8l
        device: PCIe
    ```
[(3)](https://docs.frigate.video/configuration/object_detectors)

**Intel Arc (A380) and OpenVINO:**
- Intel Arc GPUs and recent Intel CPUs/IGPUs are supported via OpenVINO.
- Inference times on an Arc A380 are around 6–10 ms per frame for common models[(1)](https://docs.frigate.video/frigate/hardware).
- OpenVINO supports a wide range of model architectures (YOLOv3/v4/v7/v9, YOLO-NAS, RF-DETR, etc.)[(4)](https://docs.frigate.video/configuration/object_detectors).

**Model Types:**
- The "model" in Frigate refers to the neural network used for detection (e.g., YOLOv6n, SSDLite MobileNet v2, YOLO-NAS). Different models vary in speed, accuracy, and hardware compatibility.
- Coral devices use TensorFlow Lite models (e.g., SSDLite MobileNet v2). Hailo uses models compiled to .hef format (e.g., YOLOv6n). OpenVINO/Arc can use ONNX/IR models, supporting various architectures[(4)](https://docs.frigate.video/configuration/object_detectors).
- You can specify a custom model in your detector configuration if you want to try different architectures[(4)](https://docs.frigate.video/configuration/object_detectors).

**Face Detection:**
- Object detection (person, car, etc.) uses the accelerator (Coral/Hailo/OpenVINO), but face recognition/enrichment uses the GPU for acceleration only if configured for enrichments. A Coral cannot accelerate enrichments such as face recognition[(5)](https://docs.frigate.video/configuration/hardware_acceleration_enrichments).
- If you want hardware-accelerated face recognition, you need to use an AMD, Intel, or Nvidia GPU for enrichments; Coral is only for object detection[(5)](https://docs.frigate.video/configuration/hardware_acceleration_enrichments).

**Video Decoding:**
- For 4K cameras, using a dedicated GPU for hardware video decoding is highly recommended (Intel QuickSync, Nvidia, or AMD VAAPI). Your K620 helps with decoding, but newer GPUs will provide more headroom[(6)](https://docs.frigate.video/configuration/hardware_acceleration_video).

**Summary:**
- Coral is still a strong choice for Frigate object detection, but Hailo-8L and Intel Arc A380 are viable and well-supported alternatives with similar or better performance depending on the model and workload[(1)](https://docs.frigate.video/frigate/hardware)[(4)](https://docs.frigate.video/configuration/object_detectors)[(3)](https://docs.frigate.video/configuration/object_detectors).
- Hailo-8L is affordable and supported, but availability may vary.
- Intel Arc GPUs (A380) support a broader range of models and can be used for both detection and enrichments.
- For face recognition acceleration, Coral will not help—you must use a supported GPU for that[(5)](https://docs.frigate.video/configuration/hardware_acceleration_enrichments).

If you want to switch from Coral USB to an internal card, both Hailo-8L and Intel Arc A380 are good options. The best choice depends on your model preferences, budget, and whether you want GPU-accelerated enrichments.

📚 **Sources:**
1. [Recommended hardware](https://docs.frigate.video/frigate/hardware)
2. [Supported Hardware](https://docs.frigate.video/configuration/object_detectors)
3. [Supported Hardware](https://docs.frigate.video/configuration/object_detectors)
4. [Supported Hardware](https://docs.frigate.video/configuration/object_detectors)
5. [Enrichments](https://docs.frigate.video/configuration/hardware_acceleration_enrichments)
6. [Video Decoding](https://docs.frigate.video/configuration/hardware_acceleration_video)

---

Have another question? Just tag [@inkeep](https://inkeep.com).

**Comment by NickM-27** [comment]
The AI response is more general, I will try to answer your question specifically.

The coral is limited because it can only run one model at a time. Devices like an Arc A380 can run many models and also more accurate models (like YOLO-NAS and YOLOv9). The coral still works with Frigate but it is less versatile when compared to a GPU

**Reply by foltz61** [reply]
Can anyone tell me how to get a custom model to work with the Coral? I added it to the \config folder snd pointed the model path to it with no luck. I also added a volume named \models to the container (because of bind mount reference in the docs) and put the model there. Also updating the model path to point there. Both methods appear to fall back to the built in model. Thanks for any help.

**Comment by leccelecce** [comment]
> Is there any benefit to switching to a Hailo 8L or potentially an Intel Arc A380 instead of just buying a Coral 2xM2?

What are your priorities for Frigate? Best possible accuracy? Maximum scalability? Optimised power consumption? Future-proofing?

2x M2 Coral's would be overkill for 4 cameras in all likelihood.
An Arc A380 reportedly averages 20W+, 75W peak. Corals are 5W peak (possibly lower for PCIE).
The Coral is a different class in terms of future-proofing and range of capability but if you don't need maximum accuracy or GPU-only features, perfectly reasonable.
